src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataDocumentationDaoTest.java

```diff
@@ -0,0 +1,182 @@
+import static org.sagebionetworks.bridge.dynamodb.DynamoReportDataDaoTest.OFFSET_KEY;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertSame;
+
+public class DynamoHealthDataDocumentationDaoTest {
+    private static final String DOC_ID = "test-documentation";
+
+    @Mock
+    private DynamoDBMapper mockMapper;
+
+    @InjectMocks
+    @Spy
+    private DynamoHealthDataDocumentationDao dao;
+
+    @BeforeMethod
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void createOrUpdateDocumentation_NewDoc() {
+        HealthDataDocumentation doc = new DynamoHealthDataDocumentation();
+
+        HealthDataDocumentation returnedDoc = dao.createOrUpdateDocumentation(doc);
+        assertSame(returnedDoc, doc);
+        assertNotNull(returnedDoc.getIdentifier());
+        assertNotNull(returnedDoc.getCreatedOn());
+
+        verify(mockMapper).save(doc);
+    }
+
+    @Test
+    public void createOrUpdateDocumentation_UpdateDoc() {
+        HealthDataDocumentation doc = new DynamoHealthDataDocumentation();
+        doc.setIdentifier(DOC_ID);
+
+        HealthDataDocumentation returnedDoc = dao.createOrUpdateDocumentation(doc);
+        assertSame(returnedDoc, doc);
+        assertEquals(doc.getIdentifier(), DOC_ID);
+        assertNotNull(returnedDoc.getModifiedOn());
+
+        verify(mockMapper).save(doc);
+    }
+
+    @Test
+    public void deleteDocumentationForParentId() {
+        // mock dependencies
+        List<DynamoHealthDataDocumentation> docList = ImmutableList.of(new DynamoHealthDataDocumentation());
+        doReturn(docList).when(dao).queryHelper(any());
+
+        // execute
+        dao.deleteDocumentationForParentId(TEST_APP_ID);
+
+        // validate
+        ArgumentCaptor<DynamoDBQueryExpression<DynamoHealthDataDocumentation>> queryCaptor = ArgumentCaptor.forClass(
+                DynamoDBQueryExpression.class);
+        verify(dao).queryHelper(queryCaptor.capture());
+
+        DynamoDBQueryExpression<DynamoHealthDataDocumentation> query = queryCaptor.getValue();
+        assertEquals(query.getHashKeyValues().getParentId(), TEST_APP_ID);
+        assertNull(query.getRangeKeyConditions());
+    }
+
+    @Test
+    public void deleteDocumentationForParentID_NoDoc() {
+        // mock dependencies
+        doReturn(ImmutableList.of()).when(dao).queryHelper(any());
+
+        // execute
+        dao.deleteDocumentationForParentId(TEST_APP_ID);
+
+        // query is validated, just validate that we don't call batchDelete()
+        verify(mockMapper, never()).batchDelete(anyList());
+    }
+
+    @Test
+    public void getDocumentationForIdentifier() {
+        DynamoHealthDataDocumentation doc = new DynamoHealthDataDocumentation();
+        when(mockMapper.load(any(DynamoHealthDataDocumentation.class))).thenReturn(doc);
```

Comment:
We should use an ArgumentCaptor and verify the doc ID and parent ID.

Reply:
I'm not sure what you mean here. Do we need an ArgumentCaptor<String> to verify that hashkey.setIdentifier() and hashkey.setParentId() are called? I'm not sure how to verify what hashkey calls inside the dao function.